community/kernel/src/main/java/org/neo4j/kernel/NeoStoreDataSource.java

```diff
@@ -534,6 +533,34 @@ public void start() throws IOException
         databaseHealth.healed();
     }
 
+    private static RecordFormats findDesiredStoreFormats( Config config, File storeDir, FileSystemAbstraction fs,
+            PageCache pageCache, LogService logService )
+    {
+        RecordFormats result;
```

Comment:
Record format selection logic should not be here, it should be moved to RecordFormatSelector